src/services/routeOptimizer.js

```
@@ -348,51 +397,98 @@ export const optimizeRoute = async (waypoints, options = {}) => {
+        totalDistance: totalDist + returnDist,
+        returnDistance: returnDist,
+        clusterHomePoint,
+      }]
+    }
+    flights.push(...clusterFlights)
   }
 
+  // routeIndices: orderedWaypoints の元waypoints配列内インデックス
+  // O(n) で構築（findIndex の O(n²) を回避）
+  const idToIndex = new Map(waypoints.map((wp, i) => [wp.id, i]))
+  const routeIndices = orderedWaypoints.map((wp) => idToIndex.get(wp.id) ?? -1)
```

ESLint

[error] 'routeIndices' is assigned a value but never used. Allowed unused vars must match /^[A-Z_]/u.

(no-unused-vars)